4.5 release page (#1138)
godotengine/godot-website: collections/_release_4_5/entry-general-2d-chunk-tilemap-physics.md

---
type: entry
section: general
subsection: _2D
rank: 0
importance: 2
anchor: chunk-tilemap-physics
title: Chunk tilemap physics
blockquote: Fusion… HA!
text: |

>   Ever since 2D tilemaps were first implemented, physics always relied on the concept of “one tile, one body”. While it works for most games, it can easily lead to performance issues in 2D scenes relying on physics as the number of bodies is extremely wasteful.

Reviewer: ```suggestion
  Ever since 2D tilemaps were first implemented, physics always relied on the concept of “one tile, one body”. While this works for most games, it can easily lead to performance issues in 2D scenes relying on physics as the number of bodies is extremely wasteful.
```

>   Since the implementation of 2D tilemaps in Godot, physics always relied on the concept of "one tile, one body". While it works for most games, it can easily lead to performance issues on 2D scenes relying on physics as the number of bodies is extremely wasteful.

Reviewer 2: ```suggestion
  Since 2D tilemaps were first implemented in Godot, physics always relied on the concept of "one tile, one body". While it works for most games, it can easily lead to performance issues in 2D scenes relying on physics as the number of bodies is extremely wasteful.
```
I feel like there might be a better way to word the ending as well ("...as the number of bodies is extremely wasteful"), but nothing specific is coming to mind for me at the moment.

Reviewer 3: I'd make it "ever since", otherwise it feels like "because they were implemented"

>   The system has been reworked from the ground-up. Now, `TileMapLayer` physics merge cell shapes into bigger collision shapes, whenever possible.

Reviewer 3: ```suggestion
  The system has been reworked from the ground up. Now, `TileMapLayer` physics merge cell shapes into bigger collision shapes, whenever possible.
```
No hyphen here

>   The system has been reworked from the ground up. Now, `TileMapLayer` physics merge cell shapes into bigger collision shapes, whenever possible.

Reviewer 2: ```suggestion
  The system has been reworked from the ground up. Now, `TileMapLayer` physics merge cell shapes into bigger collision shapes whenever possible.
```
